server/DataAccess/Common/Repositories/BaseRepository.cs

```diff
@@ -0,0 +1,58 @@
+using CoffeeShop.BusinessLogic.Common.Entities;
+using CoffeeShop.BusinessLogic.Common.Interfaces;
+using Microsoft.EntityFrameworkCore;
+
+namespace CoffeeShop.DataAccess.Common.Repositories;
+
+public abstract class BaseRepository<TEntity> : IBaseRepository<TEntity>
+    where TEntity : BaseEntity
+{
+    protected readonly DbSet<TEntity> DbSet;
+
+    protected BaseRepository(ApplicationDbContext dbContext)
+    {
+        DbSet = dbContext.Set<TEntity>();
+    }
+
+    public void Add(TEntity entity)
+    {
+        DbSet.Add(entity);
+    }
+
+    public async Task<TEntity> Get(Guid id)
+    {
+        var entity = await DbSet.FindAsync(id);
+        if (entity is not null)
+        {
+            return entity;
+        }
+
+        throw new KeyNotFoundException($"Entity with ID {id} was not found.");
```

Comment:
Galim pridet custom exception `EntityNotFoundException`, kad galetume handleinti middleware po to?